platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClient.java

```diff
@@ -0,0 +1,353 @@
+        }
+
+        return soapElement.getOwnerDocument();
+    }
+
+    /**
+     * Creates a SOAP message of the AttributeQuery request.
+     *
+     * @param attributeQuery is added to the SOAP message
+     * @return soapElement is the Element of the SOAP message
+     */
+    private Element createSOAPPMessage(AttributeQuery attributeQuery)
+            throws AttributeQueryException {
+        LOGGER.debug("Creating SAML SOAP object.");
+
+        XMLObjectBuilderFactory builderFactory = Configuration.getBuilderFactory();
+        BodyBuilder soapBodyBuilder = (BodyBuilder) builderFactory
+                .getBuilder(Body.DEFAULT_ELEMENT_NAME);
+
+        EnvelopeBuilder soapEnvelopeBuilder = (EnvelopeBuilder) builderFactory
+                .getBuilder(Envelope.DEFAULT_ELEMENT_NAME);
+
+        HeaderBuilder soapHeaderBuilder = (HeaderBuilder) builderFactory
+                .getBuilder(Header.DEFAULT_ELEMENT_NAME);
+
+        Body body = soapBodyBuilder.buildObject();
+        body.getUnknownXMLObjects().add(attributeQuery);
+
+        Envelope envelope = soapEnvelopeBuilder.buildObject();
+        envelope.setBody(body);
+
+        Header header = soapHeaderBuilder.buildObject();
+        envelope.setHeader(header);
+
+        LOGGER.debug("SAML SOAP object created.");
+
+        try {
+            return new EnvelopeMarshaller().marshall(envelope);
+        } catch (MarshallingException e) {
+            throw new AttributeQueryException("Cannot marshall SOAP object to an Element.", e);
+        }
+    }
+
+    /**
+     * Sends the request to the external attribute store via an HTTPS Client.
+     *
+     * @param requestDocument of the request
+     * @return Assertion of the response
+     * @throws AttributeQueryException
+     */
+    private Assertion sendRequest(Document requestDocument) throws AttributeQueryException {
+        URL url;
+        HttpsURLConnection connection;
+        DataOutputStream write;
+        try {
+            url = new URL(externalAttributeStoreUrl);
+
+            connection = createHttpsUrlConnection(url);
+            connection.setRequestMethod("POST");
+            connection.setRequestProperty("Content-Type", "text/xml");
+            connection.setRequestProperty("SOAPAction",
+                    "http://www.oasis-open.org/committees/security");
+            connection.setRequestProperty("Cache-Control", "no-cache, no-store");
+            connection.setRequestProperty("Pragma", "no-cache");
+            connection.setDoOutput(true);
+
+            TransformerProperties transformerProperties = new TransformerProperties();
+            transformerProperties.addOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+            write = new DataOutputStream(connection.getOutputStream());
+            write.writeBytes(XMLUtils.format(requestDocument, transformerProperties));
+            write.flush();
+            write.close();
```

Comment:
If you were to throw an IOException on line 209, you'd leave this stream open. Better to create it with try-with-resources right here:

```
try (OutputStream write = new DataOutputStream(connection.getOutputStream())) {
...
}